client.go

```diff
@@ -295,6 +296,63 @@ func (c *client) UpdateLRPInstance(
 
 	// We are assuming that a 404 means that the rep has not updated
 	// yet, but will roll soon.  This is for backwards compatibility.
+	if resp.StatusCode == http.StatusNotFound {
+		logger.Error("failed-with-status", err, lager.Data{"status-code": resp.StatusCode, "msg": http.StatusText(resp.StatusCode)})
+		// The v1 UpdateLRPInstance route is only for updating InternalRoutes
+		// on old versions of rep. This is for backwards compatibility.
+		if update.InternalRoutes != nil {
```

Comment:
This is the change that really prevents overwriting internal routes in the `new bbs, old rep` case.

Reply:
Makes sense, just saw the implementation of [executor.Update](https://github.com/cloudfoundry/executor/blob/update-metric-tags/depot/containerstore/storenode.go#L598-L619) that it handles both cases.